### PR TITLE
A0-3150: bump to 1.0.0

### PR DESCRIPTION
**Pull request by kostekIV**
# Description
Bump substrate to 1.0.0 branch.

* also bump of subxt was needed to 0.30
* bump of nightly toolchain used in aux crates

todo: test contracts migrations

## Type of change

Please delete options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as

### CODE REVIEW

File: bin/cliain/rust-toolchain.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "nightly-2023-01-10"
+channel = "nightly-2023-05-22"
```
Comment (Marcin-Radecki) [inline bin/cliain/rust-toolchain.toml:2]: Is it possible to use some stable version? Like 1.70.X ?

File: bin/node/src/cli.rs [hunk 2]
```diff
@@ -60,10 +60,6 @@ impl SubstrateCli for Cli {
         };
         Ok(Box::new(chainspec?))
     }
-
-    fn native_runtime_version(_: &Box<dyn ChainSpec>) -> &'static RuntimeVersion {
```
Comment (Marcin-Radecki) [inline bin/node/src/cli.rs:64]: What was this call used for?

File: bin/runtime/src/lib.rs [hunk 3]
```diff
@@ -229,10 +229,10 @@ impl pallet_balances::Config for Runtime {
     type ExistentialDeposit = ExistentialDeposit;
     type AccountStore = System;
     type WeightInfo = pallet_balances::weights::SubstrateWeight<Runtime>;
-    type HoldIdentifier = ();
     type FreezeIdentifier = ();
     type MaxHolds = ConstU32<0>;
     type MaxFreezes = ConstU32<0>;
+    type RuntimeHoldReason = ();
```
Comment (Marcin-Radecki) [inline bin/runtime/src/lib.rs:235]: Any impact of pallet balance functionality? Ie what this one is intened for

File: bin/runtime/src/lib.rs [hunk 4]
```diff
@@ -568,10 +568,10 @@ impl pallet_staking::Config for Runtime {
     type BenchmarkingConfig = StakingBenchmarkingConfig;
     type WeightInfo = PayoutStakersDecreasedWeightInfo;
     type CurrencyBalance = Balance;
-    type OnStakerSlash = NominationPools;
     type HistoryDepth = HistoryDepth;
     type TargetList = pallet_staking::UseValidatorsMap<Self>;
     type AdminOrigin = EnsureRoot<AccountId>;
+    type EventListeners = NominationPools;
```
Comment (Marcin-Radecki) [inline bin/runtime/src/lib.rs:574]: Same as above, what this one is intended for?

File: scripts/update_substrate_git_branch.sh [hunk 5]
```diff
@@ -28,7 +28,7 @@ for path in ${paths[@]}; do
     sed -e '/https:\/\/github.com\/Cardinal-Cryptography\/substrate\(.git\)\{0,1\}"/s/\(branch\s*=\s*"\)[^"]*"\([^,}]*\)/\1'"${BRANCH//\//\\/}"'"\2/' < $path > x
     mv x "${path}"
 
-    cargo update --manifest-path "${path}"
+    # cargo update --manifest-path "${path}"
```
Comment (Marcin-Radecki) [inline scripts/update_substrate_git_branch.sh:31]: Is it intended to be commented out?

**Review by Marcin-Radecki — APPROVED** [review]
Mostly questions.

**Review by lesniak43 — APPROVED** [review]

File: Cargo.lock [hunk 6]
```diff
@@ -2846,41 +2836,41 @@ checksum = "6c2141d6d6c8512188a7891b4b01590a45f6dac67afb4f255c4124dbb86d4eaa"
...
- "sp-application-crypto 7.0.0 (git+https://github.com/Cardinal-Cryptography/substrate.git?branch=aleph-v0.9.43)",
- "sp-core 7.0.0 (git+https://github.com/Cardinal-Cryptography/substrate.git?branch=aleph-v0.9.43)",
- "sp-io 7.0.0 (git+https://github.com/Cardinal-Cryptography/substrate.git?branch=aleph-v0.9.43)",
- "sp-runtime 7.0.0 (git+https://github.com/Cardinal-Cryptography/substrate.git?branch=aleph-v0.9.43)",
- "sp-runtime-interface 7.0.0 (git+https://github.com/Cardinal-Cryptography/substrate.git?branch=aleph-v0.9.43)",
- "sp-std 5.0.0 (git+https://github.com/Cardinal-Cryptography/substrate.git?branch=aleph-v0.9.43)",
- "sp-storage 7.0.0 (git+https://github.com/Cardinal-Cryptography/substrate.git?branch=aleph-v0.9.43)",
+ "sp-api 4.0.0-dev (git+https://github.com/Cardinal-Cryptography/substrate.git?branch=aleph-v1.0.0)",
+ "sp-application-crypto 23.0.0",
+ "sp-core 21.0.0",
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply Cargo.lock:2852]: Not sure about the 14 major version bump but this dependency is taken from the github.

File: bin/cliain/rust-toolchain.toml (hunk 1 above)
Comment (kostekIV) [reply bin/cliain/rust-toolchain.toml:2]: nope, aleph-client uses features from nightly only

File: bin/runtime/src/lib.rs (hunk 4 above)
Comment (kostekIV) [reply bin/runtime/src/lib.rs:574]: No functional changes, see https://github.com/paritytech/substrate/pull/14527 for more info

File: scripts/update_substrate_git_branch.sh (hunk 5 above)
Comment (kostekIV) [reply scripts/update_substrate_git_branch.sh:31]: Nope

File: bin/node/src/cli.rs (hunk 2 above)
Comment (kostekIV) [reply bin/node/src/cli.rs:64]: I will be honest Im not sure exactly how it was used :P

File: bin/runtime/src/lib.rs [hunk 7]
```diff
@@ -720,6 +720,11 @@ impl pallet_contracts::Config for Runtime {
     type MaxStorageKeyLen = ConstU32<128>;
     type UnsafeUnstableInterface = ConstBool<false>;
     type MaxDebugBufferLen = ConstU32<{ 2 * 1024 * 1024 }>;
+    type Migrations = (
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply bin/runtime/src/lib.rs]: yep, removed

File: bin/node/src/cli.rs (hunk 2 above)
Comment (Marcin-Radecki) [reply bin/node/src/cli.rs:64]: Let's run featurnet from this branch and leave till tomorrow. If all fine then I'll resolve this conversation.

**Pull request merged**